packages/connect/src/connector/smoldot-light.ts

```diff
@@ -134,6 +135,8 @@ export interface Config {
    * value will be used.
    */
   maxLogLevel?: number
+
+  portToWorkerFactory?: () => MessagePort
```

Comment:
I was thinking that perhaps it would be slightly better to have something like:
```ts
workerFactory?: () => Worker
```

and then on the `start` method we could do something like:

```ts
const newClientPromise = getStart().then((start) => {
  let portToWorker: MessagePort | undefined = undefined
  if (config.workerFactory) {
    const { port1, port2 } = new MessageChannel()
    config.workerFactory().postMessage(port1, [port1])
    portToWorker = port2
  }
  start({
    portToWorker,
    ...otherStuff
  });
})
```

That way the public API would be a lot simpler. Using the example of the description of the PR, it would look like this:

```ts
createScClient({
  embeddedNodeConfig: {
    workerFactory: () => new ScWorker()
  },
})
```

WDYT? 🤔

Reply:
yes this is a better public API